src/app/cli/src/coda_main.ml

```diff
@@ -121,77 +97,283 @@ module type Kernel_intf = sig
+end
+
+module Ledger_builder_hash = struct
+  include Ledger_builder_hash.Stable.V1
+
+  let ledger_hash = Ledger_builder_hash.ledger_hash
+
+  let of_aux_and_ledger_hash = Ledger_builder_hash.of_aux_and_ledger_hash
+end
+
+module Ledger_hash = struct
+  include Ledger_hash.Stable.V1
+
+  let to_bytes = Ledger_hash.to_bytes
+end
 
-  val conf_dir : string
+module Frozen_ledger_hash = struct
+  include Frozen_ledger_hash.Stable.V1
 
-  val lbc_tree_max_depth : [`Infinity | `Finite of int]
+  let to_bytes = Frozen_ledger_hash.to_bytes
 
-  val propose_keypair : Keypair.t option
+  let of_ledger_hash = Frozen_ledger_hash.of_ledger_hash
+end
 
-  val genesis_proof : Snark_params.Tock.Proof.t
+module User_command = struct
+  include (
+    User_command :
+      module type of User_command
+      with module With_valid_signature := User_command.With_valid_signature )
 
-  val transaction_capacity_log_2 : int
-  (** Capacity of transactions per block *)
+  let fee (t : t) = Payload.fee t.payload
 
-  val commit_id : Client_lib.Git_sha.t option
+  let sender (t : t) = Public_key.compress t.sender
 
-  val work_selection : Protocols.Coda_pow.Work_selection.t
-end
+  let seed = Secure_random.string ()
 
-module type Work_selector_F = functor
-  (Inputs : Work_selector.Inputs.Inputs_intf)
-  -> Protocols.Coda_pow.Work_selector_intf
-     with type ledger_builder := Inputs.Ledger_builder.t
-      and type work :=
-                 ( Inputs.Ledger_proof_statement.t
-                 , Inputs.Transaction.t
-                 , Inputs.Sparse_ledger.t
-                 , Inputs.Ledger_proof.t )
-                 Snark_work_lib.Work.Single.Spec.t
-      and type snark_pool := Inputs.Snark_pool.t
-      and type fee := Inputs.Fee.t
+  let compare t1 t2 = User_command.Stable.V1.compare ~seed t1 t2
 
-module type Init_intf = sig
-  include Config_intf
+  module With_valid_signature = struct
+    module T = struct
+      include User_command.With_valid_signature
 
-  include Kernel_intf
+      let compare t1 t2 = User_command.With_valid_signature.compare ~seed t1 t2
+    end
 
-  module Make_work_selector : Work_selector_F
+    include T
+    include Comparable.Make (T)
+  end
+end
 
-  val proposer_prover : [`Proposer of Prover.t | `Non_proposer]
+module Ledger_proof_statement = Transaction_snark.Statement
+module Completed_work =
+  Ledger_builder.Make_completed_work (Public_key.Compressed) (Ledger_proof)
+    (Ledger_proof_statement)
 
-  val verifier : Verifier.t
+module Ledger_builder_diff = Ledger_builder.Make_diff (struct
+  module Ledger_proof = Ledger_proof
+  module Ledger_hash = Ledger_hash
+  module Ledger_builder_hash = Ledger_builder_hash
+  module Ledger_builder_aux_hash = Ledger_builder_aux_hash
+  module Compressed_public_key = Public_key.Compressed
+  module User_command = User_command
+  module Completed_work = Completed_work
+end)
 
-  val genesis_proof : Proof.t
-end
+module Consensus_mechanism = Consensus.Mechanism.Make (Ledger_builder_diff)
```

Comment:
I won't require it for this PR, but you should be able to kill all of these functors as well. If you decide this and aren't sure how, I can give you some advice.